oarepo_ui/theme/assets/semantic-ui/js/oarepo_ui/forms/hooks.js

```diff
@@ -22,45 +25,210 @@ export const useVocabularyOptions = (vocabularyType) => {
+      // save accepts posts/puts even with validation errors. Here I check if there are some errors in the response
+      // body. Here I am setting the individual error messages to the field
+      if (response.errors) {
+        response.errors.forEach((error) =>
+          setFieldError(error.field, error.messages[0])
+        );
+        // here I am setting the state to be used by FormFeedback componene that plugs into the formik's context.
+        // note that we are using removeNullAndInternalFields to clean this up before submitting
+        setFieldValue("validationErrors", {
+          errors: response.errors,
+          errorMessage: i18next.t(
+            "Draft saved with validation errors. Fields listed below that failed validation were not saved to the server"
+          ),
+        });
+        return false;
+      }
+      setFieldValue("successMessage", i18next.t("Draft saved successfully."));
+      return response;
+    } catch (error) {
+      // handle 400 errors. Normally, axios would put messages in error.response. But for example
+      // offline Error message does not produce a response, so in this way we can display
+      // network error message
+      setFieldValue(
+        "httpErrors",
+        error?.response?.data?.message ?? error.message
+      );
+      return false;
+    } finally {
+      setSubmitting(false);
+    }
+  }
+
+  async function publish() {
+    // call save and if save returns false, exit
+    const saveResult = await save();
+    if (!saveResult) return;
+    // imperative form validation, if fails exit
+    const validationErrors = await validateForm();
+    if (!_isEmpty(validationErrors)) return;
+    setSubmitting(true);
+    let response;
+    try {
+      response = await apiClient.publishDraft(saveResult);
+
+      window.location.href = response.links.self_html;
+      setFieldValue(
+        "successMessage",
+        i18next.t(
+          "Draft published successfully. Redirecting to record's detail page ..."
+        )
+      );
+
+      return response;
+    } catch (error) {
+      // in case of validation errors on the server during publish, in RDM they return a 400 and below
+      // error message. Not 100% sure if our server does the same.
+      if (
+        error?.response &&
+        error.response.data?.status === 400 &&
+        error.response.data?.message === "A validation error occurred."
+      ) {
+        error.errors?.forEach((err) =>
+          setFieldError(err.field, err.messages.join(" "))
+        );
+      } else {
+        setFieldValue(
+          "httpErrors",
+          error?.response?.data?.message ?? error.message
+        );
+      }
+
+      return false;
+    } finally {
+      setSubmitting(false);
+    }
+  }
+
+  async function _delete() {
+    setSubmitting(true);
+
+    try {
+      let response = await apiClient.deleteDraft(values);
+
+      // TODO: should redirect to /me page in user dashboard?? but we don't have that one yet
```

Comment:
We should consider adding redirect url as function parameter here, there shouldn't be any hardcoded `/docs/` redirect url as this is just a generic ui library. I'll update the code here accordingly

Reply:
@mirekys Ahoj, agreed, it was just a temporary measure. In RMD, they hardcode url as /me/uploads, which takes the user back to their dashboard. I was thinking that some similar link, would come inside of formConfig, as it is not exactly a link that should be delivered within the record's links. However, do I understand well, that you wish for _delete function in useDepositApiClient to accept a redirection url param and to redirect to that page after deletion is done?

Reply:
Yes, exactly, and then we can pass it whatever we want, it will be specific for each instance of form & repository. For draft-enabled repositories, it will be `/user/<record-model>` (e.g. `/user/docs/`), for vocabularies, it could be just `/vocabularies/` or parent vocabulary item detail url...